src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -5874,6 +5886,63 @@ private ScanTypeFlags ScanTupleType(out SyntaxToken lastTokenOfType)
             return ScanTypeFlags.NotType;
         }
 
+        private ScanTypeFlags ScanFunctionPointerType(out SyntaxToken lastTokenOfType)
+        {
+            Debug.Assert(IsFunctionPointerStart());
+            _ = EatContextualToken(SyntaxKind.DelegateKeyword);
+            _ = EatToken(SyntaxKind.AsteriskToken);
+
+            switch (CurrentToken.ContextualKind)
+            {
+                case SyntaxKind.CdeclKeyword:
+                case SyntaxKind.ManagedKeyword:
+                case SyntaxKind.StdcallKeyword:
+                case SyntaxKind.ThiscallKeyword:
+                case SyntaxKind.UnmanagedKeyword:
+                    _ = EatContextualToken(CurrentToken.ContextualKind);
+                    break;
+
+                default:
+                    if (CurrentToken.Kind != SyntaxKind.LessThanToken)
+                    {
+                        _ = EatToken();
```

Comment:
>EatToken [](start = 28, length = 8)

ScanType is supposed to scan the tokens to see if we have a type.  If we don't have a less-than token, then we don't have a type and this should return an appropriate flag saying so.

Reply:
Well, `delegate*` can _only_ mean one thing: someone was trying to write a function pointer. That's why I went with this form.

---
In reply to: [348132938](https://github.com/dotnet/roslyn/pull/39879#discussion_r348132938) [](ancestors = 348132938)

Reply:
I agree with @333fred on that point.  It's an unambiguous prefix for a func-ptr type.  So i think once that part is scanned, it's fine to say that we're still in "we're seeing a type" mode.